resources/js/views/settings/roles/View.vue

```diff
@@ -0,0 +1,294 @@
+      <b-container fluid>
+        <b-form-group
+          label-cols-sm='3'
+          :label="$t('settings.roles.name')"
+          label-for='name'
+          :state='fieldState("name")'
+        >
+          <b-form-input id='name' type='text' v-model='model.name' :state='fieldState("name")' :disabled='isBusy || viewOnly'></b-form-input>
+          <template slot='invalid-feedback'><div v-html="fieldError('name')"></div></template>
+        </b-form-group>
+        <b-form-group
+          label-cols-sm='3'
+          :label="$t('settings.roles.roomLimit.label')"
+          label-for='room-limit'
+          :state='fieldState("room_limit")'
+        >
+          <b-form-input id='room-limit' type='number' :state='fieldState("room_limit")' v-model='model.room_limit' min='-1' :disabled='isBusy || viewOnly || model.room_limit === null || model.room_limit === -1'></b-form-input>
+          <b-form-checkbox
+            :checked='model.room_limit == null'
+            @change='(checked) => checked ? model.room_limit = null : model.room_limit = 0'
+            :disabled='isBusy || viewOnly'
+            inline
+          >
+            {{ $t('settings.roles.roomLimit.default', { value: settings('room_limit') === -1 ? $t('settings.roles.roomLimit.unlimited').toLowerCase() : settings('room_limit') }) }}
+          </b-form-checkbox>
+          <b-form-checkbox
+            :checked='model.room_limit == -1'
+            @change='(checked) => checked ? model.room_limit = -1 : model.room_limit = 0'
+            :disabled='isBusy || viewOnly'
+            inline
+          >
+            {{ $t('settings.roles.roomLimit.unlimited') }}
+          </b-form-checkbox>
+          <template slot='invalid-feedback'><div v-html="fieldError('room_limit')"></div></template>
+        </b-form-group>
+        <b-form-group
+          :label="$t('settings.roles.permissions')"
+          label-size='lg'
+          label-class='font-weight-bold pt-0'
+          :state="Object.keys(errors).some(error => error === 'permissions' || error.startsWith('permissions.')) ? false : null"
+        >
+          <b-overlay :show='permissionsLoading'>
+            <div class='row' v-if='!permissionsLoading && Object.keys(permissions).length > 0'>
+              <b-form-group
+                class='col-lg-4 col-sm-12'
+                v-for="key in Object.keys(permissions)"
+                :key='key'
+              >
+                <template v-slot:label>
+                  <b>{{ $t(`app.permissions.${key}.title`) }}</b>
+                </template>
+
+                <b-form-checkbox-group
+                  v-model='model.permissions'
+                  :options='permissions[key]'
+                  stacked
+                  text-field='translatedName'
+                  value-field='id'
+                  :disabled='isBusy || viewOnly'
+                  :state="Object.keys(errors).some(error => error === 'permissions' || error.startsWith('permissions.')) ? false : null"
```

Comment:
why are you not using the mixins helper?

Reply:
Because this is a special case that isn't supported by the mixins helper